plugins/factory-plugin/src/workspace-manager.ts

```diff
@@ -0,0 +1,48 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as theia from '@theia/plugin';
+import * as che from '@eclipse-che/plugin';
+import { che as cheApi } from '@eclipse-che/api';
+
+/**
+ * Make checks on workspace configuration and shows dedicated information to user.
+ */
+export class WorkspaceManager {
```

Comment:
I wouldn't call it `WorkspaceManager` because it is too generic. `Manager` usually is bad naming. Maybe `WorkspaceEphemeralChecker` ? or do you have other suggestion ?

Reply:
Maybe then EphemeralWorkspaceChecker?

Reply:
+1 on `EphemeralWorkspaceChecker`